lib/Dialect/TritonGPU/Transforms/Pipeliner/PipelineExpander.cpp

```diff
@@ -248,7 +281,9 @@ bool LoopPipelinerInternal::verifySchedule() {
         continue;
       int64_t producerCycle = it->second;
       if (consumerCycle < producerCycle - numCylesPerIter * distance) {
-        consumer->emitError("operation scheduled before its operands");
+        LDBG("--operation scheduled before its operands: " << *consumer
+                                                           << " ->  BAIL");
+        PERF_WARNING(*consumer, "operation scheduled before its operands");
```

Comment:
I downgrade it because the test input is after all valid Triton code and Triton should generate some code as a compiler, although the performance could be bad. I don't know if emitting an error would stop compilation, but in general I feel error means the code itself is incorrect.

How about reporting in the warning that the compiler has some limitations and we will address them? For example, I file an issue about the limitation here, and include the issue number in the warning message?

Reply:
Oh I don't think we need to go that far to mention an issue in the warning message. Just wondering if you have looked at the reason why we have operation scheduled before operands for the test case and if it is possible to dump more information in the warning.

Reply:
I see. Sorry for the confusion! Updated.

Reply:
Yes, for this test input, I think printing the producer is enough to demonstrate the cause. I just updated the code to include a remark on the producer.
A good error message should include the consumer and producer in the same message. Limited by the error reporting API of MLIR, I cannot make better error messages. I can make a customized error reporting in future PRs.